app/billing_client.go

```diff
@@ -0,0 +1,310 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	pb "go.viam.com/api/app/v1"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// UsageCostType specifies the type of usage cost.
+type UsageCostType int32
+
+const (
+	// UsageCostTypeUnspecified is an unspecified usage cost type.
+	UsageCostTypeUnspecified UsageCostType = iota
+	// UsageCostTypeDataUpload represents the usage cost from data upload.
+	UsageCostTypeDataUpload
+	// UsageCostTypeDataEgress represents the usage cost from data egress.
+	UsageCostTypeDataEgress
+	// UsageCostTypeRemoteControl represents the usage cost from remote control.
+	UsageCostTypeRemoteControl
+	// UsageCostTypeStandardCompute represents the usage cost from standard compute.
+	UsageCostTypeStandardCompute
+	// UsageCostTypeCloudStorage represents the usage cost from cloud storage.
+	UsageCostTypeCloudStorage
+	// UsageCostTypeBinaryDataCloudStorage represents the usage cost from binary data cloud storage.
+	UsageCostTypeBinaryDataCloudStorage
+	// UsageCostTypeOtherCloudStorage represents the usage cost from other cloud storage.
+	UsageCostTypeOtherCloudStorage
+	// UsageCostTypePerMachine represents the usage cost per machine.
+	UsageCostTypePerMachine
+)
+
+func usageCostTypeFromProto(costType pb.UsageCostType) UsageCostType {
+	switch costType {
+	case pb.UsageCostType_USAGE_COST_TYPE_UNSPECIFIED:
+		return UsageCostTypeUnspecified
+	case pb.UsageCostType_USAGE_COST_TYPE_DATA_UPLOAD:
+		return UsageCostTypeDataUpload
+	case pb.UsageCostType_USAGE_COST_TYPE_DATA_EGRESS:
+		return UsageCostTypeDataEgress
+	case pb.UsageCostType_USAGE_COST_TYPE_REMOTE_CONTROL:
+		return UsageCostTypeRemoteControl
+	case pb.UsageCostType_USAGE_COST_TYPE_STANDARD_COMPUTE:
+		return UsageCostTypeStandardCompute
+	case pb.UsageCostType_USAGE_COST_TYPE_CLOUD_STORAGE:
+		return UsageCostTypeCloudStorage
+	case pb.UsageCostType_USAGE_COST_TYPE_BINARY_DATA_CLOUD_STORAGE:
+		return UsageCostTypeBinaryDataCloudStorage
+	case pb.UsageCostType_USAGE_COST_TYPE_OTHER_CLOUD_STORAGE:
+		return UsageCostTypeOtherCloudStorage
+	case pb.UsageCostType_USAGE_COST_TYPE_PER_MACHINE:
+		return UsageCostTypePerMachine
+	default:
+		return UsageCostTypeUnspecified
+	}
+}
+
+// UsageCost contains the cost and cost type.
+type UsageCost struct {
+	ResourceType UsageCostType
+	Cost         float64
+}
+
+func usageCostFromProto(cost *pb.UsageCost) *UsageCost {
+	return &UsageCost{
+		ResourceType: usageCostTypeFromProto(cost.ResourceType),
+		Cost:         cost.Cost,
+	}
+}
+
+// ResourceUsageCosts holds the usage costs with discount information.
+type ResourceUsageCosts struct {
```

Comment:
just want to flag this struct and all the other structs below if we decide to move all the structs to the top of the file. I think we are having this convo in a few diff places lol so I will just slack you and we can talk about it 😄